VocaDbModel/Service/VideoServices/VideoServiceSoundCloud.cs

```diff
@@ -73,7 +77,25 @@ VideoUrlParseResult ReturnError(Exception x, string additionalInfo = null)
 
 			try
 			{
-				result = await JsonRequest.ReadObjectAsync<SoundCloudResult>(apiUrl, timeout: TimeSpan.FromSeconds(10));
+				// Code from: https://github.com/TerribleDev/OwinOAuthProviders/blob/8b382b0429aeb656f54149ab6f3472dd559ae12f/src/Owin.Security.Providers.SoundCloud/SoundCloudAuthenticationHandler.cs#L74
+				// TODO: use IHttpClientFactory
+				using var httpClient = new HttpClient();
+				var tokenResponse = await httpClient.PostAsync("https://api.soundcloud.com/oauth2/token", new FormUrlEncodedContent(new List<KeyValuePair<string?, string?>>
+				{
+					new("client_id", apikey),
+					new("client_secret", AppConfig.SoundCloudClientSecret),
+					new("grant_type", "client_credentials"),
+				}));
+				tokenResponse.EnsureSuccessStatusCode();
+				var text = await tokenResponse.Content.ReadAsStringAsync();
+
+				var response = JsonConvert.DeserializeObject<SoundCloudTokenResponse>(text);
+				var accessToken = response.Access_token;
+
+				result = await JsonRequest.ReadObjectAsync<SoundCloudResult>(apiUrl, timeout: TimeSpan.FromSeconds(10), headers: headers =>
+				{
+					headers.Authorization = new AuthenticationHeaderValue("OAuth", accessToken);
+				});

```

Comment:
We should check if `result` is null, otherwise a `NullPointerException` will be thrown later in this code.